docs/markdown/authoritative/backend-generic-sql.md

```diff
@@ -0,0 +1,211 @@
+# Generic SQL Backends
+The generic SQL backends (like gmysql, gpgsql and godbc) are backends with easily
+configurable SQL statements, allowing you to graft PowerDNS on any SQL database
+of your choosing. Because all database schemas will be different, a generic
+backend is needed to cover all needs.
+
+**Warning**: Host names and the MNAME of a SOA records are NEVER terminated with
+a '.' in PowerDNS storage! If a trailing '.' is present it will inevitably cause
+problems, problems that may be hard to debug.
+
+# Basic functionality
+4 queries are needed for regular lookups and 1 is needed for zone transfers.
+
+The 4+4 regular queries must return the following 8 fields, in this exact order:
+
+* content: This is the 'right hand side' of a DNS record. For an A record, this is the IP address for example.
+* ttl: TTL of this record, in seconds. Must be a real value, no checking is performed.
+* prio: For MX records, this should be the priority of the mail exchanger specified.
+* qtype: The ASCII representation of the qtype of this record. Examples are 'A', 'MX', 'SOA', 'AAAA'. Make sure that this field returns an exact answer - PowerDNS won't recognise 'A ' as 'A'. This can be achieved by using a VARCHAR instead of a CHAR.
+* domain\_id: Each domain must have a unique domain\_id. No two domains may share a domain\_id, all records in a domain should have the same. A number.
```

Comment:
positive number. also IIRC domain_id must be unique across all backends, not just each backend